### PR TITLE
CommonAI interface

### PR DESCRIPTION
**Pull request by MartinLundDam**
**WHAT IS IT?**
- Interface that exposes the pathfinding method that has its functionality in the AStarAI module. 

**MISSING:** 
- The method {nextCoordinateInPath} still needs to be overwritten in the AStarAI module in order to function.

### CODE REVIEW

File: CommonAI/src/main/java/com/github/sef24sp4/common/ai/IPathfindingProvider.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.github.sef24sp4.common.ai;
+
+import com.github.sef24sp4.common.entities.IEntity;
+import com.github.sef24sp4.common.interfaces.IVector;
+
+public interface IPathfindingProvider {
+
+	/**
+	 *
```
Comment (MidnightRocket) [inline CommonAI/src/main/java/com/github/sef24sp4/common/ai/IPathfindingProvider.java:9]: Missing a General description of what the method does.

**Review by MidnightRocket — CHANGES_REQUESTED** [review]

File: CommonAI/src/main/java/com/github/sef24sp4/common/ai/IPathfindingProvider.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.github.sef24sp4.common.entities.IEntity;
+import com.github.sef24sp4.common.interfaces.IVector;
+
+public interface IPathfindingProvider {
+
+	/**
+	 * Entity can by implementing this method navigtate to a TargatCoordinate.
+	 * The Method start from the StartCoordinate of the Entity, open neighbor nodes, and expand the node with the lowest fCost.
+	 * When the best route towards a target is known. The Entity will navigate to the next step on this route, which is the TargatCoordinate.
```
Comment (FearfulSnow) [inline CommonAI/src/main/java/com/github/sef24sp4/common/ai/IPathfindingProvider.java:11]: The text should not have any info regarting the inner workings of the specific AI, if ALL AI's use nodes and fCost, then i am wrong. 
rewrite: Get the next coordinate of a valid path between entity and target coordinate

File: CommonAI/src/main/java/com/github/sef24sp4/common/ai/IPathfindingProvider.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	 * @param entity is for now a PolygonEnemy.
+	 * @param targetCoordinate is the next step for PolygonEnemy to go to.
+	 * @return an IVector, that is the next node (targetCoordinate), that the entity should go to. This can be a coordinate.
+	 */
+	public IVector calculateNextMove(IEntity entity, IVector targetCoordinate);
```
Comment (FearfulSnow) [inline CommonAI/src/main/java/com/github/sef24sp4/common/ai/IPathfindingProvider.java:16]: Possible name changes:
nextCoordinateInPath()
getFirstCoordinateInPath()

**Review by FearfulSnow — CHANGES_REQUESTED** [review]

**Review by FearfulSnow — APPROVED** [review]
good ready to be merged

**Review by MidnightRocket — APPROVED** [review]

**Review by FearfulSnow — APPROVED** [review]

**Review by MidnightRocket — APPROVED** [review]

**Pull request merged**